### ISSUE TITLE
Side Chat

### ISSUE THREAD

**Issue by satoshipuzzles** [vicariousdrama/cornychat#27]
Chats within a chat.

- Labels for each side chat 
- Volume control for each chat.
- stream audio in side chats
- record side chats

**Comment by satoshipuzzles** [comment]
<img width="420" alt="Screenshot 2024-03-19 at 5 27 43 PM" src="https://github.com/vicariousdrama/cornychat/assets/127057721/e612cda9-ab78-4268-a629-2bb45c5827a2">

**Comment by TekkadanPlays** [comment]
![sidechat](https://github.com/vicariousdrama/cornychat/assets/93434084/aa5ab6c6-2b61-4628-a55b-6f3f5e2d7972)

- Main chat would initially have a green "+" button
- Sidechat would be created beneath mainchat, a "v" or down arrow would become available in main chat to move the user into the new room
- Sidechat would have an "^" or up arrow to move the user back to main chat
- If a sidechat is empty the "sidechat" is deleted

**Comment by vicariousdrama** [comment]
Should all other rooms with people in them show for discoverability?

**Comment by TekkadanPlays** [comment]
> Should all other rooms with people in them show for discoverability?

No, forcing people to check the main page creates a positive feedback loop. It creates an association between "typing cornychat.com into my browser" and seeing which rooms are currently open. 

More rooms = more dopamine

More users = more dopamine

You create FOMO with statistical engagement.

Having sidechats enables the rooms to support a higher user count per room. This will encourage people to customize their rooms for future use, or use when the owner is not available. It creates "lifespans" of rooms. It enables people to categorize their discussions and abandon them when they desire. Much like a social butterfly flows from group to group at a party.

**Comment by TekkadanPlays** [comment]
Just putting this out there but I think there is still plenty of benefit to sidechats even if there is no ability to "listen in" on other chats within the room. I think it is preferable if possible, but the underlying mechanic would still create a more flexible space for sharing conversation.

**Comment by TekkadanPlays** [comment]
Would like to formally add the suggestion to be able to "invite another user to a sidechat".

I think in terms of how the UI might work, a small request should appear above or below the navbar at the bottom. 

"User x has invited you to [name of sidechat]. Accept/Decline"

I think the addition of sidechats and the social effort to establish their use would enable groups to summon particular users into their conversation, which would be great for when users need to segregate their conversations within a room.

![New Project(2)](https://github.com/vicariousdrama/cornychat/assets/93434084/c48b3255-836a-4604-88a3-48788fced839)
